tools/ai_sidecar/server.py

```
@@ -97,6 +119,88 @@ async def _safe_send(websocket: Any, payload: dict[str, Any]) -> None:
+            )
+        logger.info(
+            "ws upgrade accepted client=%s peer=%s",
+            client_id,
+            getattr(connection, "remote_address", None),
+        )
+        return None
+
+    return _check
+
+
+async def _broadcast_external(frame: dict[str, Any], *, exclude: Any) -> None:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `exclude`

(ANN401)